test/src/main/java/org/apache/accumulo/test/TableOperationsIT.java

```diff
@@ -189,6 +190,22 @@ public void createTable() throws TableExistsException, AccumuloException,
     accumuloClient.tableOperations().delete(tableName);
   }
 
+  @Test
+  public void createTableWithTableNameLengthLimit()
+      throws AccumuloException, AccumuloSecurityException, TableExistsException {
+    StringBuilder tableNameBuilder = new StringBuilder();
+    for (int i = 0; i <= MAX_TABLE_NAME_LEN; i++) {
+      tableNameBuilder.append('a');
+    }
+    String tableName = tableNameBuilder.toString();
+    try {
+      accumuloClient.tableOperations().create(tableName);
+      fail("IllegalArgumentException was not thrown");
+    } catch (IllegalArgumentException exc) {
+      assertTrue(!accumuloClient.tableOperations().exists(tableName));
+    }
```

Comment:
These can be simplified like:

```suggestion
    assertThrows(IllegalArgumentException.class, () -> accumuloClient.tableOperations().create(tableName));
    assertFalse(accumuloClient.tableOperations().exists(tableName));
```

The `Assert.assertThrows` also returns the exception, so you can do additional checks to check the exception's `getMessage()` and, if you wish, the `getCause()`.

You can also create the name using commons-lang3's `String tableName = StringUtils.repeat('a', MAX_TABLE_NAME_LEN);`

You can also verify the length of the string in an assert statement. Additional checks would include max length plus and minus one.

These comments also apply to the namespace checks.